YesGraph/yesgraph-superwidget
# Test options

### What’s this PR do?
- Reorganizes our tests so that we can test multiple options & configurations for the Superwidget
- Uses browserify to enable ES6 syntax in our tests
- Displays an error message when oauth succeeds but contact importing fails
- Code cleanup in karma.conf.js

### Any background context you want to provide?
We've had bugs arise in the past that only occur with certain Superwidget options, or with certain HTML flags set. In our automated tests, we need to check for these types of bugs as well.

To do that, we need to be able to run the same tests multiple times with different Superwidget HTML snippets (called "fixtures").

That was impossible with our previous setup, because we were specifying the test scripts & fixtures in the configuration file:
```javascript
/* Snippet from the old version of karma.conf.js */
files: [
    // Dependencies
    'http://code.jquery.com/jquery-2.1.1.min.js',
    'https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.5.8/clipboard.min.js',

    // Fixtures
    'tests/fixtures.html',

    // TestS
    'tests/*.js',

    // Source code
    'dist/dev/yesgraph-invites.js',      
 ]
```
It's now made possible by listing only "tests/master.js" in the configuration file, and then importing & running our test suites with each of the necessary fixtures from _within_ "tests/master.js".

### Where should the reviewer start?
Start by reviewing tests/master.js

### Does this require changes on the API side?
Nope

### How should this be manually tested?
First run `npm install` to get the Browserify, then run the tests with `gulp test`. The output should be something like this:

<img width="674" alt="screen shot 2017-01-06 at 2 03 07 pm" src="https://cloud.githubusercontent.com/assets/10701968/21734593/0a047c2c-d419-11e6-906f-735d8e379798.png">

### What are the relevant tickets?
[Asana: Test multiple options & configurations for the Superwidget](https://app.asana.com/0/59202558034519/228324330390285)

### Does the knowledge base need an update?
Nope
